File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/ViewModel/BrowseCardsViewModel.java
package com.shuneault.netrunnerdeckbuilder.ViewModel;

import com.shuneault.netrunnerdeckbuilder.db.CardRepository;
import com.shuneault.netrunnerdeckbuilder.game.CardList;
import com.shuneault.netrunnerdeckbuilder.game.CardPool;
import com.shuneault.netrunnerdeckbuilder.game.Format;
import com.shuneault.netrunnerdeckbuilder.helper.ISettingsProvider;

import java.util.ArrayList;

import androidx.lifecycle.ViewModel;

import org.jetbrains.annotations.NotNull;

public class BrowseCardsViewModel extends ViewModel {
    private CardRepository cardRepo;
    private CardPool mPool;
    public ArrayList<String> packFilter = new ArrayList<>();
    private CardList mCards;
    private ISettingsProvider settingsProvider;
    public String searchText = "";
    public Format browseFormat;

    public BrowseCardsViewModel(CardRepository cardRepo, ISettingsProvider settingsProvider) {
        this.cardRepo = cardRepo;
        browseFormat = cardRepo.getDefaultFormat();
        this.settingsProvider = settingsProvider;
    }

    public void init() {
        if (mPool == null){
            mPool = cardRepo.getCardPool();
            mCards = mPool.getCards();
        }
    }

    public CardList getCardList() {
        return mCards;
    }

    public void doSearch(String searchText) {
        mCards.clear();
        mCards.addAll(cardRepo.searchCards(searchText, mPool));
    }

    public void updatePackFilter(ArrayList<String> packFilter) {
        mPool = cardRepo.getCardPool(this.browseFormat, packFilter);
        this.packFilter = packFilter;
    }

    public void useMyCollectionAsFilter() {
        packFilter = settingsProvider.getMyCollection();
    }

    public void setFilter(@NotNull ArrayList<String> values) {
        packFilter = values;
    }
}

File: app/src/main/java/com/shuneault/netrunnerdeckbuilder/ViewModel/BrowseCardsViewModel.kt
package com.shuneault.netrunnerdeckbuilder.ViewModel

import androidx.lifecycle.ViewModel
import com.shuneault.netrunnerdeckbuilder.db.CardRepository
import com.shuneault.netrunnerdeckbuilder.game.CardList
import com.shuneault.netrunnerdeckbuilder.game.CardPool
import com.shuneault.netrunnerdeckbuilder.game.Format
import com.shuneault.netrunnerdeckbuilder.helper.ISettingsProvider
import java.util.*

class BrowseCardsViewModel(private val cardRepo: CardRepository, settingsProvider: ISettingsProvider) : ViewModel() {
    private var mPool: CardPool = cardRepo.cardPool
    var packFilter = ArrayList<String>()
    var cardList: CardList? = null
        private set
    private val settingsProvider: ISettingsProvider
    var searchText = ""
    var browseFormat: Format
    fun init() {
        cardList = mPool.getCards()
    }

    fun doSearch(searchText: String?) {
        cardList!!.clear()
        cardList!!.addAll(cardRepo.searchCards(searchText, mPool))
    }

    fun updatePackFilter(packFilter: ArrayList<String>, format: Format) {
        browseFormat = format
        this.packFilter = packFilter
        mPool = cardRepo.getCardPool(format, packFilter)
    }

    fun useMyCollectionAsFilter() {
        updatePackFilter(settingsProvider.myCollection, browseFormat)
    }

    init {
        browseFormat = cardRepo.defaultFormat
        this.settingsProvider = settingsProvider
    }
}